### ISSUE TITLE
Docker hungs in docker 1.13.1

### ISSUE THREAD

**Issue by msavlani** [moby/moby#35176]
**Description**
We have docker 1.13.1 installed in our environment and we see docker hung's indefinitely.
Its installed on physical machine with Redhat OS 7.3 having kernel 3.10.

Below is the go routine stack trace  --
[goroutine-stacks-2017-10-11T110718-0500.log](https://github.com/moby/moby/files/1377583/goroutine-stacks-2017-10-11T110718-0500.log)

We see this issue every couple of days.

**Steps to reproduce the issue:**
1. Restart docker 1.13.1 
2. Wait for couple of days.
3. Then do docker ps

 **Go routine stack Stack** 
Below is go routine stack trace -- I have also attached full stack trace with this issue.

goroutine 1 [chan receive, 1992 minutes]:
main.(*DaemonCli).start(0xc420275fb0, 0x0, 0x1908125, 0x17, 0xc42038f000, 0xc4201a3130, 0xc42049e990, 0x0, 0x0)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/cmd/dockerd/daemon.go:326 +0x183f
main.runDaemon(0x0, 0x1908125, 0x17, 0xc42038f000, 0xc4201a3130, 0xc42049e990, 0x20, 0x0)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/cmd/dockerd/docker.go:86 +0xb2
main.newDaemonCommand.func1(0xc42039d680, 0xc4203f1c00, 0x0, 0x20, 0x0, 0x0)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/cmd/dockerd/docker.go:42 +0x71
github.com/docker/docker/vendor/github.com/spf13/cobra.(*Command).execute(0xc42039d680, 0xc42000c490, 0x20, 0x23, 0xc42039d680, 0xc42000c490)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/github.com/spf13/cobra/command.go:646 +0x26d
github.com/docker/docker/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42039d680, 0x16f5620, 0xc4201a3101, 0xc42042fb20)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/github.com/spf13/cobra/command.go:742 +0x377
github.com/docker/docker/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42039d680, 0xc42042fb20, 0xc42000e0b8)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/github.com/spf13/cobra/command.go:695 +0x2b
main.main()
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/cmd/dockerd/docker.go:106 +0xe2

goroutine 17 [syscall, 1992 minutes, locked to thread]:
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 18 [syscall]:
os/signal.signal_recv(0x24318a0)
        /usr/local/go/src/runtime/sigqueue.go:116 +0x157
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.1
        /usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 99 [syscall, 1992 minutes]:
syscall.Syscall6(0xf7, 0x1, 0x4f82, 0xc4201aa080, 0x1000004, 0x0, 0x0, 0x424c95, 0xc4201a30b8, 0x422439)
        /usr/local/go/src/syscall/asm_linux_amd64.s:44 +0x5
os.(*Process).blockUntilWaitable(0xc4201969f0, 0xc42000f9d8, 0xc4201a30c0, 0xc420275c10)
        /usr/local/go/src/os/wait_waitid.go:28 +0xbc
os.(*Process).wait(0xc4201969f0, 0x0, 0x422439, 0xc42000f9d8)
        /usr/local/go/src/os/exec_unix.go:22 +0xab
os.(*Process).Wait(0xc4201969f0, 0x4184eb, 0xc42000f9d8, 0x0)
        /usr/local/go/src/os/doc.go:49 +0x2b
os/exec.(*Cmd).Wait(0xc420328420, 0x1a0bfb0, 0x0)
        /usr/local/go/src/os/exec/exec.go:434 +0x6d
github.com/docker/docker/libcontainerd.(*remote).runContainerdDaemon.func1(0xc420328420, 0xc420495e10)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/libcontainerd/remote_unix.go:430 +0x2b
created by github.com/docker/docker/libcontainerd.(*remote).runContainerdDaemon
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/libcontainerd/remote_unix.go:432 +0xe02
goroutine 102 [chan receive]:
github.com/docker/docker/libcontainerd.(*remote).handleConnectionChange(0xc420495e10)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/libcontainerd/remote_unix.go:147 +0xf7
created by github.com/docker/docker/libcontainerd.New
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/libcontainerd/remote_unix.go:121 +0x5d2

goroutine 101 [select, 1992 minutes]:
github.com/docker/docker/vendor/google.golang.org/grpc.(*addrConn).transportMonitor(0xc4203286e0)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/google.golang.org/grpc/clientconn.go:747 +0x63e
github.com/docker/docker/vendor/google.golang.org/grpc.(*ClientConn).resetAddrConn.func1(0xc4203286e0)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/google.golang.org/grpc/clientconn.go:506 +0x1de
created by github.com/docker/docker/vendor/google.golang.org/grpc.(*ClientConn).resetAddrConn
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/google.golang.org/grpc/clientconn.go:507 +0x3c3

goroutine 104 [chan receive]:
github.com/docker/docker/daemon/graphdriver/devmapper.(*DeviceSet).startDeviceDeletionWorker(0xc420508b60)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/daemon/graphdriver/devmapper/deviceset.go:702 +0x100
created by github.com/docker/docker/daemon/graphdriver/devmapper.(*DeviceSet).initMetaData
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/daemon/graphdriver/devmapper/deviceset.go:734 +0x21f

goroutine 115 [IO wait]:
net.runtime_pollWait(0x7f79d74962e8, 0x72, 0x7)
        /usr/local/go/src/runtime/netpoll.go:160 +0x59
net.(*pollDesc).wait(0xc4203d4140, 0x72, 0xc42020bb18, 0xc420010178)
        /usr/local/go/src/net/fd_poll_runtime.go:73 +0x38
net.(*pollDesc).waitRead(0xc4203d4140, 0x242c220, 0xc420010178)
        /usr/local/go/src/net/fd_poll_runtime.go:78 +0x34
net.(*netFD).Read(0xc4203d40e0, 0xc420098000, 0x8000, 0x8000, 0x0, 0x242c220, 0xc420010178)
        /usr/local/go/src/net/fd_unix.go:243 +0x1a1
net.(*conn).Read(0xc42007c000, 0xc420098000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/net.go:173 +0x70
bufio.(*Reader).fill(0xc4203a8300)
        /usr/local/go/src/bufio/bufio.go:97 +0x10c
bufio.(*Reader).Read(0xc4203a8300, 0xc4203a20f8, 0x9, 0x9, 0x469210, 0xc43126e400, 0xc42020bd18)
        /usr/local/go/src/bufio/bufio.go:209 +0x1bc
io.ReadAtLeast(0x24224a0, 0xc4203a8300, 0xc4203a20f8, 0x9, 0x9, 0x9, 0xc435d003f8, 0x2425da0, 0xc435d00600)
        /usr/local/go/src/io/io.go:307 +0xa4
io.ReadFull(0x24224a0, 0xc4203a8300, 0xc4203a20f8, 0x9, 0x9, 0xc61e91, 0xc42866a3c0, 0x0)
        /usr/local/go/src/io/io.go:325 +0x58
github.com/docker/docker/vendor/golang.org/x/net/http2.readFrameHeader(0xc4203a20f8, 0x9, 0x9, 0x24224a0, 0xc4203a8300, 0x0, 0xc400000000, 0xc420330280, 0xc)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/golang.org/x/net/http2/frame.go:236 +0x7b
github.com/docker/docker/vendor/golang.org/x/net/http2.(*Framer).ReadFrame(0xc4203a20c0, 0x0, 0x0, 0x0, 0x0)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/golang.org/x/net/http2/frame.go:463 +0xa4
github.com/docker/docker/vendor/google.golang.org/grpc/transport.(*framer).readFrame(0xc420015d70, 0xc435d00570, 0xc435d00570, 0x0, 0x0)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/google.golang.org/grpc/transport/http_util.go:508 +0x2f
github.com/docker/docker/vendor/google.golang.org/grpc/transport.(*http2Client).reader(0xc4204e4100)
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/google.golang.org/grpc/transport/http2_client.go:922 +0xb6
created by github.com/docker/docker/vendor/google.golang.org/grpc/transport.newHTTP2Client
        /root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/vendor/google.golang.org/grpc/transport/http2_client.go:203 +0x8b1
:

**Output of `docker version`:**

```
$ docker version
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:38:28 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.1
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:38:28 2017
 OS/Arch:      linux/amd64
 Experimental: false

```

**Output of `docker info`:**

```
Containers: 177
 Running: 108
 Paused: 0
 Stopped: 69
Images: 301
Server Version: 1.13.1
Storage Driver: devicemapper
 Pool Name: docker-thinpool
 Pool Blocksize: 524.3 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file:
 Metadata file:
 Data Space Used: 66.89 GB
 Data Space Total: 204 GB
 Data Space Available: 137.1 GB
 Metadata Space Used: 21.86 MB
 Metadata Space Total: 2.143 GB
 Metadata Space Available: 2.121 GB
 Thin Pool Minimum Free Space: 20.4 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.135-RHEL7 (2016-09-28)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: aa8187dbd3b7ad67d8e5e3a15115d3eef43a7ed1
runc version: 9df8b306d01f59d3a8029be411de015b7304dd8f
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-514.26.1.el7.x86_64
Operating System: Red Hat Enterprise Linux
OSType: linux
Architecture: x86_64
CPUs: 64
Total Memory: 502.9 GiB
Name: node101
ID: EM7W:DWO6:QNGK:RCHN:2CWF:DWCQ:XCE4:XBEQ:ZEM4:ZKER:BJSR:QDRY
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Experimental: false
Insecure Registries:
 172.30.0.0/16
 127.0.0.0/8
Live Restore Enabled: false

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

We are using redhat 7.3

$ cat /etc/os-release
NAME="Red Hat Enterprise Linux Server"
VERSION="7.3 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="7.3"
PRETTY_NAME="Red Hat Enterprise Linux"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.3:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.3
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.3"

**Comment by thaJeztah** [comment]
Docker 1.13.1 Community Edition has reached end of life; If you're still encountering this on a current version of Docker for RHEL, could you open a new issue with details in the https://github.com/docker/for-linux issue tracker?

I'm closing this issue for now for housekeeping, but feel free to comment

**Comment by msavlani** [comment]
Hi thaJeztah,

I know this is has reached end of cycle. But unfortunately we are in production with this version of docker and cannot upgrade unless we pass whole testing cycle.

Can someone just anaylze my go routine log and provide some input as what might be causing this issue ?

@rhvgoyal  @cpuguy83 @mlaventure 

Can you guys help me here ?
Any help is appreciated.

Thanks,
Manish Sawlani

**Comment by cpuguy83** [comment]
@msavlani 


```
goroutine 952068 [semacquire, 541 minutes]:
sync.runtime_Semacquire(0xc420438994)
	/usr/local/go/src/runtime/sema.go:47 +0x30
sync.(*Mutex).Lock(0xc420438990)
	/usr/local/go/src/sync/mutex.go:85 +0xd0
github.com/docker/docker/daemon/logger/jsonfilelog.(*JSONFileLogger).Close(0xc420438980, 0x1, 0x1)
	/root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/daemon/logger/jsonfilelog/jsonfilelog.go:138 +0x4b
github.com/docker/docker/container.(*Container).Reset(0xc421ef3e00, 0xc425344d00)
	/root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/container/monitor.go:42 +0x201
github.com/docker/docker/daemon.(*Daemon).StateChanged(0xc42045e800, 0xc4261c2500, 0x40, 0xc4214d4808, 0x4, 0x8f00000000, 0x0, 0x0, 0x0, 0x0, ...)
	/root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/daemon/monitor.go:43 +0x2dc
github.com/docker/docker/libcontainerd.(*container).handleEvent.func1()
	/root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/libcontainerd/container_unix.go:217 +0x82
github.com/docker/docker/libcontainerd.(*queue).append.func1(0xc421f21d00, 0x0, 0xc426450280, 0xc42539b200)
	/root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/libcontainerd/queue_unix.go:28 +0x30
created by github.com/docker/docker/libcontainerd.(*queue).append
	/root/rpmbuild/BUILD/docker-engine/.gopath/src/github.com/docker/docker/libcontainerd/queue_unix.go:30 +0x170

```

This is fixed probably even in Docker CE 17.03.2. Given that there are no new features in the 17.03 as compared to 1.13, I'd recommend you at least upgrade to that, however the current supported release is 17.09.
